cfdm/read_write/netcdf/netcdfwrite.py

```diff
@@ -10,11 +10,16 @@
 
 from .. import IOWrite
 
+from .netcdfread import NetCDFRead
+
+from ...cfdmimplementation import implementation
 from ...decorators import _manage_log_level_via_verbosity
 
 
 logger = logging.getLogger(__name__)
 
+_implementation = implementation()
```

Comment:
Better to use self.implementation, i.e. effective_fields = NetCDFRead(self.implementation).read(filename), as this will play nicely with downstream applications (i.e. cf.python)
```suggestion
```